modules/exoplanets/tasks/exoplanetsdatapreparationtask.cpp

```diff
@@ -0,0 +1,449 @@
+
+namespace {
+    constexpr const char* KeyInputDataFile = "InputDataFile";
+    constexpr const char* KeyInputSpeck = "InputSPECK";
+    constexpr const char* KeyOutputBin = "OutputBIN";
+    constexpr const char* KeyOutputLut = "OutputLUT";
+    constexpr const char* KeyTeffToBv = "TeffToBvFile";
+
+    constexpr const char* _loggerCat = "ExoplanetsDataPreparationTask";
+} // namespace
+
+namespace openspace::exoplanets {
+
+ExoplanetsDataPreparationTask::ExoplanetsDataPreparationTask(const ghoul::Dictionary& dictionary) {
+    openspace::documentation::testSpecificationAndThrow(
+        documentation(),
+        dictionary,
+        "ExoplanetsDataPreparationTask"
+    );
+
+    _inputDataPath = absPath(dictionary.value<std::string>(KeyInputDataFile));
+    _inputSpeckPath = absPath(dictionary.value<std::string>(KeyInputSpeck));
+    _outputBinPath = absPath(dictionary.value<std::string>(KeyOutputBin));
+    _outputLutPath = absPath(dictionary.value<std::string>(KeyOutputLut));
+    _teffToBvFilePath = absPath(dictionary.value<std::string>(KeyTeffToBv));
+}
+
+std::string ExoplanetsDataPreparationTask::description() {
+    return fmt::format(
+        "Extract data about exoplanets from file '{}' and write as bin to '{}'. The data "
+        "file should be a csv version of the Planetary Systems Composite Data from the "
+        "NASA exoplanets archive (https://exoplanetarchive.ipac.caltech.edu/).",
+        _inputDataPath, _outputBinPath
+    );
+}
+
+void ExoplanetsDataPreparationTask::perform(const Task::ProgressCallback& progressCallback) {
+    std::ifstream inputDataFile(_inputDataPath);
+    if (!inputDataFile.good()) {
+        LERROR(fmt::format("Failed to open input file '{}'", _inputDataPath));
+        return;
+    }
+
+    std::ofstream binFile(_outputBinPath, std::ios::out | std::ios::binary);
+    std::ofstream lutFile(_outputLutPath);
+
+    int version = 1;
+    binFile.write(reinterpret_cast<char*>(&version), sizeof(int));
+
+    std::string planetRow;
+    getline(inputDataFile, planetRow); // The first line, containing the data names
+
+    // Read column names into a vector, for later access
+    std::vector<std::string> columnNames;
+    std::stringstream sStream(planetRow);
+    std::string colName;
+    while (getline(sStream, colName, ',')) {
+        columnNames.push_back(colName);
+    }
+
+    // Read total number of items
+    int total = 0;
+    while (getline(inputDataFile, planetRow)) {
+        ++total;
+    }
+    inputDataFile.clear();
+    inputDataFile.seekg(0);
+    getline(inputDataFile, planetRow); // The first line, containing the data names
+
+    LINFO(fmt::format("Loading {} exoplanets", total));
+
+    auto readFloatData = [](const std::string& str) -> float {
+    #ifdef WIN32
+        float result;
+        auto [p, ec] = std::from_chars(str.data(), str.data() + str.size(), result);
+        if (ec == std::errc()) {
+            return result;
+        }
+        return NAN;
```

Comment:
Instead of using the C style NAN, which is a `#define` in <cmath>, it might be better to use `std::numeric_limits<float>::quiet_NaN()` instead?
It might be very much more verbose, but more accuratly reflect that it's an "allowed" NaN, rather than a `signaling_NaN`